### PR TITLE
fix: Updated set_null() function to be more useful #6030

### PR DESCRIPTION
**Pull request by laf**
DO NOT DELETE THIS TEXT

#### Please note

> Please read this information carefully. You can run `./scripts/pre-commit.php` to check your code before submitting.

- [x] Have you signed the [Contributors agreement](http://docs.librenms.org/General/Contributing/) - please do NOT submit a pull request unless you have (signing the agreement in the same pull request is fine). Your commit message for signing the agreement must appear as per the docs.
- [x] Have you followed our [code guidelines?](http://docs.librenms.org/Developing/Code-Guidelines/)

#### Testers

If you would like to test this pull request then please run: `./scripts/github-apply <pr_id>`, i.e `./scripts/github-apply 5926`

Fixes: #6030 (hopefully).

@murrant ?

### CODE REVIEW

**Comment by laf** [comment]
Yes and no.

It was supposed to drop `!isset($value) || ` so 0 would be valid but I mustn't have pushed my second commit. I'll do it later tonight.

**Comment by laf** [comment]
@murrant Pushed. I also added a unit test

File: tests/CommonFunctionsTest.php [hunk 1]
```diff
@@ -83,4 +83,12 @@ public function testRrdDescriptions()
         $data = 'Toner, S/N:CR_UM-16021314488.';
         $this->assertEquals('Toner, S/N CR_UM-16021314488.', safedescr($data));
     }
+
+    public function testSetNull()
+    {
```
Comment (murrant) [inline tests/CommonFunctionsTest.php]: Needs some more tests for default and min:
```
 $this->assertEquals(99, set_null(' ', 99));
$this->assertNull(set_null(-25, null, 0));
$this->assertEquals(2, set_null(2, 0, 2));
```

**Comment by laf** [comment]
Updated

**Pull request merged**